### PR TITLE
Remove surelog libs out of own folder, use soversion

### PR DESCRIPTION
**Pull request by timkpaine**
Parallel of https://github.com/chipsalliance/UHDM/pull/1002
- install into `lib` instead of `lib/surelog`, python files moved to `lib/surelog-python` (open to other options, should probably move to a real python package)
- use `SOVERSION` so that lib/libsurelog.so is now a symlink, and we install a versioned lib/libsurelog.so.<major>.<minor>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -127,7 +127,7 @@ endif()
 
 # NOTE: Set the global output directories after the subprojects have had their go at it
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
```
[the remark this answers is not in the document]
Comment (timkpaine) [reply CMakeLists.txt]: I will test windows shortly

**Comment by alaindargelas** [comment]
yosys-systenverilog-plugin test is faiing here, can't merge.

**Comment by timkpaine** [comment]
> yosys-systenverilog-plugin test is faiing here, can't merge.

I can take a look, though it seems unrelated to this PR

**Comment by alaindargelas** [comment]
@timkpaine Looks like this is the root cause of the failure of the plugin:
https://github.com/chipsalliance/Surelog/actions/runs/6059758147

**Comment by timkpaine** [comment]
@alaindargelas pretty sure its this: https://github.com/chipsalliance/systemverilog-plugin/commit/ec729d507784b264e3c9993384b5ec9ab375e8e4

@kbieganski

**Comment by alaindargelas** [comment]
@timkpaine correct. So you can make this PR official.

**Pull request merged**